### PR TITLE
Move TemplateValidator and NodeLabeller to pointers

### PR DESCRIPTION
**Pull request by lyarwood**
**What this PR does / why we need it**:

As these attributes are optional and decorated with omitempty we can move both to be pointers so they have an empty value of nil and are omitted correctly.

**Which issue(s) this PR fixes**: 

Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by 0xFelix** [comment]
Does this API change require changes in e.g. HCO too?

File: api/v1beta1/ssp_types.go [hunk 1]
```diff
@@ -56,13 +56,13 @@ type NodeLabeller struct {
 // SSPSpec defines the desired state of SSP
 type SSPSpec struct {
 	// TemplateValidator is configuration of the template validator operand
-	TemplateValidator TemplateValidator `json:"templateValidator,omitempty"`
+	TemplateValidator *TemplateValidator `json:"templateValidator,omitempty"`
```
Comment (akrejcir) [inline api/v1beta1/ssp_types.go:59]: Can you add `nil` check to places where this is used?
For example here:
https://github.com/kubevirt/ssp-operator/blob/e67e36faecec9a3a45f290219e2bc2e297958870/internal/operands/template-validator/reconcile.go#L149-L154

And many places in functional tests.

File: api/v1beta1/ssp_types.go [hunk 2]
```diff
@@ -56,13 +56,13 @@ type NodeLabeller struct {
...
 
 	// CommonTemplates is the configuration of the common templates operand
 	CommonTemplates CommonTemplates `json:"commonTemplates"`
 
 	// NodeLabeller is configuration of the node-labeller operand
-	NodeLabeller NodeLabeller `json:"nodeLabeller,omitempty"`
+	NodeLabeller *NodeLabeller `json:"nodeLabeller,omitempty"`
```
[the remark this answers is not in the document]
Comment (lyarwood) [reply api/v1beta1/ssp_types.go:65]: Shouldn't we at least deprecate it first for a release? Happy to do that in a follow up.

[the remark this answers is not in the document]
Comment (ksimon1) [reply api/v1beta1/ssp_types.go:65]: The NL is deprecated for a really long time (when you look into operand code, you will see only function, which deletes every NL objects)

[the remark this answers is not in the document]
Comment (lyarwood) [reply api/v1beta1/ssp_types.go:65]: Yeah I see that in the package but it really should be documented here first in the future. I can remove it in a follow up PR, I don't think it's worth re-purposing this PR for now.

[the remark this answers is not in the document]
Comment (lyarwood) [reply api/v1beta1/ssp_types.go:65]: https://github.com/kubevirt/ssp-operator/pull/492 FWIW

**Pull request merged**